### PR TITLE
Feat/learning rate decay

### PR DESCRIPTION
**Pull request by RuanJohn**
## What?
Add the option to linearly decay the actor and critic learning rates during training. 
## Why?
Based PPO implementation details blog [here](https://iclr-blog-track.github.io/2022/03/25/ppo-implementation-details/). 
## How?
Simple utils file that creates a linear learning rate decay scheduler and passes that to the optimisers. 
## Extra
Default behaviour (constant learning rates) are retained by setting `decay_learning_rates: False` in the system configs.

### CODE REVIEW

**Review by WiemKhlifi — DISMISSED** [review]
LGTM 🛥️

File: mava/systems/ff_ippo.py [hunk 1]
```diff
@@ -355,13 +356,25 @@ def learner_setup(
     actor_network, critic_network = networks.make(
         config=config, network="feedforward", centralised_critic=False
     )
+
+    actor_lr = (
+        make_learning_rate_schedule(config.system.actor_lr, config)
```
Comment (OmaymaMahjoub) [inline mava/systems/ff_ippo.py]: how about instead of making lr scheduler you get_lr directly: by that you save the if statements to be inside the function

Comment (RuanJohn) [reply mava/systems/ff_ippo.py]: Super nice idea! What do you think of the changes?

File: mava/systems/ff_ippo.py [hunk 2]
```diff
@@ -357,16 +357,8 @@ def learner_setup(
...
-        make_learning_rate_schedule(config.system.actor_lr, config)
-        if config.system.decay_learning_rates
-        else config.system.actor_lr
-    )
-    critic_lr = (
-        make_learning_rate_schedule(config.system.critic_lr, config)
-        if config.system.decay_learning_rates
-        else config.system.critic_lr
-    )
+    actor_lr = make_learning_rate(config.system.actor_lr, config)
```
Comment (OmaymaMahjoub) [inline mava/systems/ff_ippo.py:360]: isn't sending the config enough to retrieve the actor_lr?

File: mava/systems/ff_ippo.py (hunk 1 above)
Comment (OmaymaMahjoub) [reply mava/systems/ff_ippo.py]: Yes! Thanks for the changes :)

File: mava/systems/ff_ippo.py (hunk 2 above)
Comment (RuanJohn) [reply mava/systems/ff_ippo.py:360]: It is difficult to do, since we would then have to pass in an arg to say that we want to make a scheduler for the actor or the critic. So I felt that passing in an initial learning rate keeps things as general as possible.

Comment (OmaymaMahjoub) [reply mava/systems/ff_ippo.py:360]: Oh I get it now! Thanks for the explanation :pray:

**Review by WiemKhlifi — APPROVED** [review]
Thank you for these changes @RuanJohn 🥳 🙌

**Review by OmaymaMahjoub — APPROVED** [review]
Thanks @RuanJohn!
